infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -106,6 +105,7 @@
   private final Executor executor;
   private final KubernetesRuntimeStateCache runtimeStates;
   private final KubernetesMachineCache machines;
+  private final StartSynchronizer startSynchronizer;
 
   @Inject
   public KubernetesInternalRuntime(
```

Comment:
Seems OK, but, to be honest, I no longer can validate the logic of this class as it gets too complicated. I encourage you, and myself, to refactor this class as soon as possible to simplify it drastically.